tests/other_errors/double_plusplus.jou

```diff
@@ -0,0 +1,3 @@
+def main() -> int:
+    foo = 1
+    x = &++foo  # Error: the address-of operator '&' cannot be used with the result of incrementing a value
```

Comment:
- This and `address_of_plusplus.jou` should have file names of each other
- One of the two should use `--` instead, to check that it works too.